<?xml version="1.0"?>
<launch>

    <arg name="use_rosbot" default="false"/>
    <arg name="use_gazebo" default="false"/>

    <!-- <include if="$(arg use_gazebo)" file="$(find rosbot_gazebo)/launch/maze_world.launch"/>
    <include if="$(arg use_gazebo)" file="$(find rosbot_gazebo)/launch/rosbot.launch"/>

    <param if="$(arg use_gazebo)" name="use_sim_time" value="true"/>

    <node if="$(arg use_rosbot)" pkg="rplidar_ros" type="rplidarNode" name="rplidar">
        <param name="angle_compensate" type="bool" value="true"/>
    </node>

    <node if="$(arg use_rosbot)" pkg="tutorial_pkg" type="drive_controller_node" name="drive_controller"/>

    <node if="$(arg use_rosbot)" pkg="tf" type="static_transform_publisher" name="laser_broadcaster" args="0 0 0 3.14 0 0 base_link laser_frame 100" /> -->

    <!-- <node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="teleop_twist_keyboard" output="screen"/> -->

    <!-- <node pkg="rviz" type="rviz" name="rviz"/> -->

    <!-- <node pkg="gmapping" type="slam_gmapping" name="gmapping">
        <param name="base_frame" value="base_link"/>
        <param name="odom_frame" value="odom" />
        <param name="delta" value="0.1" />
    </node> -->

    <node pkg="move_base" type="move_base" name="move_base" output="screen">

        <remap from="openni_points" to="/rtabmap/octomap_occupied_space"/>
        <remap from="map" to="/rtabmap/grid_map"/>
        <!-- <remap from="current_goal" to="/move_base_simple/goal"/> -->

        <param name="controller_frequency" value="10.0"/>
        <!-- <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS"/> -->
        <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/>

        <rosparam file="$(find visguide)/launch/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find visguide)/launch/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find visguide)/launch/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find visguide)/launch/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find visguide)/launch/trajectory_planner.yaml" command="load" />
        <rosparam file="$(find visguide)/launch/base_local_planner_params.yaml" command="load" />
    </node>

</launch>